src/sentinel.c

```diff
@@ -2123,9 +2123,9 @@ void sentinelRefreshInstanceInfo(sentinelRedisInstance *ri, const char *info) {
 
     /* Handle slaves replicating to a different master address. */
     if ((ri->flags & SRI_SLAVE) &&
-        role == SRI_SLAVE &&
+		role == SRI_SLAVE &&
         (ri->slave_master_port != ri->master->addr->port ||
-         strcasecmp(ri->slave_master_host,ri->master->addr->ip)))
+		(ri->slave_master_host && strcasecmp(ri->slave_master_host, ri->master->addr->ip))))
```

Comment:
This looks like an issue that's not specific to Windows. Could you elaborate on what the bug is? Would it make sense to submit this to antirez\redis?